### ISSUE TITLE
Carat ^ matching is not compatible with NPM prerelease exclusion

### ISSUE THREAD

**Issue by paulortman** [rbarrois/python-semanticversion#61]
The current carat match algorithm matches pre-release versions (-alpha, -beta).  NPM only considers those versions which are not pre-releases.  Given that the carat match was added explicitly to mirror the NPM extension, it seems like identical behavior is desired.

For example, search for the "babel" package from [semver](https://semver.npmjs.com/) and then use "^5.0.0" as the search term.  None of the 5.0.0-betas are selected.  Using python-semanticversion, they would match.

To fix this will require adjusting existing tests -- and therefore potentially breaking existing behavior.

**Comment by ivankravets** [comment]
Is this the same issue?
```python
import semantic_version

semantic_version.Version("3.0.0-beta.3") in semantic_version.Spec("^2.1.1")
# True , why?
```

**Comment by paulortman** [comment]
@ivankravets This does appear to be the same issue.  Using my branch of code, your test returns False.

**Comment by rbarrois** [comment]
Thanks for the bug report!

This was fixed in version 2.7.1:
```python
>>> import semantic_version
>>> semantic_version.Version("3.0.0-beta.3") in semantic_version.NpmSpec("^2.1.1")
False
```

Sorry for the delay :/
